### PR TITLE
Clean up door sync

### PR DESCRIPTION
**Pull request by AJenbo**
Just the most basic change to make it simpler to understand the door logic later when refactoring it

### CODE REVIEW

File: Source/objects.cpp [hunk 1]
```diff
@@ -1426,7 +1434,7 @@ void ActivateTrapLine(int ttype, int tid)
 	for (int i = 0; i < ActiveObjectCount; i++) {
 		Object &trap = Objects[ActiveObjects[i]];
 		if (trap._otype == ttype && trap._oVar1 == tid) {
-			trap._oVar4 = 1;
+			trap._oVar4 = DOOR_OPEN;
```
Comment (ephphatha) [inline Source/objects.cpp:1437]: This isn't a door, this is setting a trap as being tripped/activated

File: Source/objects.cpp [hunk 2]
```diff
@@ -1437,34 +1445,34 @@ void ActivateTrapLine(int ttype, int tid)
 void UpdateFlameTrap(Object &trap)
 {
 	if (trap._oVar2 != 0) {
-		if (trap._oVar4 != 0) {
+		if (trap._oVar4 != DOOR_CLOSED) {
```
Comment (ephphatha) [inline Source/objects.cpp:1448]: Same here, this function operates on a flamehole object, not a door.

Comment (AJenbo) [reply Source/objects.cpp]: Oops, lets try that again, but this time with a bit of clean up on top :D

**Review by ephphatha — APPROVED** [review]
Would prefer the door state enum to be a named and scoped enum but the usage makes that really messy 😂

**Comment by AJenbo** [comment]
yeah, I'm not even sure it is currently being used in any place where we wouldn't need to do casting, at which point it becomes pretty pointless

**Pull request merged**